cdap-ui/app/cdap/components/AbstractWidget/InputFieldDropdown/index.tsx

```diff
@@ -47,19 +57,71 @@ function getFields(schemas: IStageSchema[]) {
   return fields;
 }
 
+function containsType(types: string[], allowedTypes: string[]) {
+  if (allowedTypes.length == 0) {
+    return true;
+  }
+
+  for (const allowedType of allowedTypes) {
+    if (types.includes(allowedType)) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
 const InputFieldDropdown: React.FC<IInputFieldProps> = ({
   value,
   onChange,
   disabled,
   extraConfig,
+  widgetProps,
 }) => {
   const inputSchema = objectQuery(extraConfig, 'inputSchema');
-  const fieldValues = getFields(inputSchema);
-  const widgetProps = {
-    options: fieldValues,
-  };
 
-  return <Select value={value} onChange={onChange} widgetProps={widgetProps} disabled={disabled} />;
+  const isMultiSelect: boolean = widgetProps.multiselect || false;
+  const allowedTypes: string[] = widgetProps.allowedTypes || [];
+
+  const fieldValues = getFields(inputSchema, allowedTypes);
+
+  const newValue = value
+    .toString()
```

Comment:
This seems like a significant change. Does this still work fine with plugins like joiner?

Reply:
Validated that joiner and similar plugins still work